aspnetcore/security/authentication/scaffold-identity/consoleAddUser/Data/SeedData.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+using WebApplication1.Data;
+
+// dotnet aspnet-codegenerator razorpage -m Contact -dc ApplicationDbContext -outDir Pages\Contacts --referenceScriptLibraries
+namespace ContactManager.Data
+{
+    #region snippet
+    public static class SeedData
+    {
+        public static async Task Initialize(IServiceProvider serviceProvider, string testUserPw)
+        {
+            using (var context = new ApplicationDbContext(
+                serviceProvider.GetRequiredService<DbContextOptions<ApplicationDbContext>>()))
```

Comment:
This should use the user manager.

Reply:
Otherwise the appropriate validation is not performed.

Reply:
> This should use the user manager.

It does, see the code that follows

`            var userManager = serviceProvider.GetService<UserManager<IdentityUser>>();`

Reply:
Apparently my coffee didn't have the expected effect 😄